libenv/sysinfo.c

```diff
@@ -1461,12 +1500,29 @@ static void OSClasses(EvalContext *ctx)
 
     // Extract major version number
     char *major = NULL;
-    for (char *sp = vbuff; *sp != '\0'; sp++)
+    char *minor = NULL;
+    char *sp = vbuff;
+    for (sp; *sp != '\0'; sp++)
```

Comment:
The first expression is a statement with no effect. If you need the char pointer later you should rather do
```suggestion
    char *sp;
    for (sp = vbuff; *sp != '\0'; sp++)
```